llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -1020,12 +1020,15 @@ class LoopVectorizationCostModel {
     /// Holds the maximum number of concurrent live intervals in the loop.
     /// The key is ClassID of target-provided register class.
     SmallMapVector<unsigned, unsigned, 4> MaxLocalUsers;
-  };
 
-  /// \return Returns information about the register usages of the loop for the
-  /// given vectorization factors.
-  SmallVector<RegisterUsage, 8>
-  calculateRegisterUsage(ArrayRef<ElementCount> VFs);
+    /// Check if any of the tracked live intervals exceeds the number of
+    /// available registers for the target.
+    bool exceedsMaxNumRegs(const TargetTransformInfo &TTI) {
```

Comment:
```suggestion
    bool exceedsMaxNumRegs(const TargetTransformInfo &TTI) const {
```

possible to use `const` here?